.github/workflows/unittests.yml

```diff
@@ -28,3 +28,11 @@ jobs:
         make check
     - name: Run unit tests
       run: make test
+    - name: Generate artifacts
+      run: ./contrib/scripts/release.sh
+    - name: Upload artifacts
```

Comment:
I understand the benefit of running the `release.sh` script but I don't understand the benefit of uploading the artifacts.
I would understand that we automatically build and upload artifacts on the release page when a release is tagged but this is not what this is doing, is it?

Reply:
This will allow us to download the binaries from each commit for testing. Which I think is a nice feature, as we might want to use these artifacts for the official releases as well.